src/Model/Payment/Braintree/CreditCard.php

```diff
@@ -29,23 +29,20 @@
 use Magento\Framework\Module\Manager;
 use Magento\Framework\Serialize\SerializerInterface;
 use Magento\Payment\Helper\Data as PaymentHelper;
-use Magento\Payment\Model\MethodInterface;
 use Magento\Sales\Api\OrderPaymentRepositoryInterface;
 use Magento\Sales\Model\Order as MagentoOrder;
 use Magento\Vault\Api\Data\PaymentTokenFactoryInterface;
 use Shopgate\Base\Api\Config\CoreInterface;
 use Shopgate\Base\Model\Shopgate\Extended\Base as ShopgateOrder;
 use Shopgate\Import\Helper\Order\Utility;
 use Shopgate\Import\Helper\Payment\Braintree as Helper;
-use Shopgate\Import\Model\Payment\AbstractPayment;
 
-class CreditCard extends AbstractPayment
+class CreditCard extends Base
```

Comment:
can u check the method `setVaultPaymentToken` in this class, says that the method does not exist. Not sure if it's because of refactoring or it was like that previously as well. Does it get set via `setData()` internally? If so, maybe needs to be set via setData to stop confusion

Reply:
Switched to setData as it will also work.